Add demonstration for robust training and robustness quantification

YuezhuXu opened the pull request:
Mainly three files: examples/tutorials/part4_Lipschitz_estimator for showing how to use package eclipse-nn;
example/control/stabilize_linear_system_with_verification for adding robustness part to Part_1_stabilize_linear_system;
example/control/ref_tracking_ODE_with_verification for adding robustness part  to Part_3_ref_tracking_ODE.

drgona committed:
hotfix missing figures for DAE example

drgona committed:
hotfix wandb inpots

YuezhuXu committed:
Create stabilize_linear_system_with_verification.ipynb

YuezhuXu committed:
some results for adding regularization in tracking controller

YuezhuXu committed:
add demo for eclipse-nn

YuezhuXu committed:
Update part_4_Lipschitz_estimator.ipynb

YuezhuXu committed:
Merge remote-tracking branch 'upstream/develop' into Lip-with-eclipse

YuezhuXu committed:
Add some demo

For robust training and robustness quantification

YuezhuXu committed:
remove obsolete plots

drgona commented on examples/tutorials/sampleweights/mat/lyr3n100test1.mat:
Ideally, we don't want to store any data files for our examples. If we need to use data from a file we want to load it from stable url. See example in this notebook: https://colab.research.google.com/github/pnnl/neuromancer/blob/master/examples/DAEs/tank_dae_example.ipynb

drgona commented on examples/tutorials/part_4_Lipschitz_estimator.ipynb:
The tutorial directory is meant to be only for base neuromancer syntax.
I suggest creating new directory called verification for the Lipschitz estimator files.
The jupyter notebooks should be didactical, the current version has minimal markdown. Please update the new version with each cell explained in markdown.

drgona reviewed (changes requested):
These are nice contributions. But we need some modificaitons before they can be merged.

Jupyter notebooks need more markdown to explain the presented methods.
[This notebook](https://github.com/YuezhuXu/neuromancer/blob/Lip-with-eclipse/examples/control/ref_tracking_ODE_with_verification.ipynb) does a good job at that, but the rest do not.

I see the training is very slow in the reference tracking ODE case with Jabocian regularizaiton, is it the same for the stabilizaiton task? 

Let's don't upload any files .mat, .npz, .pt, and others into neuromancer. For all files we should use stable url download.

drgona commented on examples/control/ref_tracking_ODE_with_verification.ipynb:
the function def make_jac_cl_node should be made way more modular. You have multiple function definitions within this function, this could be easily broken apart.

YuezhuXu committed:
vectorize in calculating jacobian

YuezhuXu committed:
Sorting

YuezhuXu committed:
Sorting

drgona committed:
acknowledgment

YuezhuXu committed:
Robustness with Lipschitz

All the files are included in a separate folder named "examples/robustness_with_Lipschitz_estimator" .
The version has solved the time issue through curriculum learning and correct vectorization in computing gradients. Code and data structure is optimized, and informative explanations are also added.

YuezhuXu committed:
Merge branch 'pnnl:master' into Lip-with-eclipse

YuezhuXu commented:
All the files are included in a separate folder named "examples/robustness_with_Lipschitz_estimator" .
The version has solved the time issue through curriculum learning and correct vectorization in computing gradients. Code and data structure is optimized, and informative explanations are also added.
